openproblems/tasks/chromatin_potential/methods/linear_regression.py

```diff
@@ -0,0 +1,189 @@
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+import scanpy as sc
+import subprocess
+from ....tools.decorators import method
+
+
+def _chrom_limit(x, tss_size=2e5):
+    """Extending TSS to upstream and downstream intervals
+
+    Parameters
+    ----------
+    x : pd.Series
+        a pd.Series containing [start, end, direction] where start and end are ints and direction is {'+', '-'}.
+    tss_size: int
+        a int that defines the upstream and downstream extending regions around TSS
+    """
+    y = x.values
+    gene_direction = y[-1]
+    gene_start = y[-3]
+    gene_end = y[-2]
+    if gene_direction == "+":
+        return [gene_start - tss_size // 2, gene_start + tss_size // 2]
+    else:
+        return [gene_end - tss_size // 2, gene_end + tss_size // 2]
+
+
+def _get_annotation(adata):
+    """This function insert meta data into .obs for adata"""
+    from pyensembl import EnsemblRelease
+
+    # TODO: add more species
```

Comment:
Is this TODO resolved? Looks like it should work with other species now.

Reply:
Yes, it should work with other species by using data loader function returns.